addons/sale/models/sale_order.py

```diff
@@ -496,6 +496,7 @@ def _compute_amounts(self):
         for order in self:
             order_lines = order.order_line.filtered(lambda x: not x.display_type)
             base_lines = [line._prepare_base_line_for_taxes_computation() for line in order_lines]
+            base_lines += order._add_base_lines_for_early_payment_discount()
```

Comment:
- `_compute_amounts` must depend of `payment_term_id`
- same for `_compute_tax_totals`